torch/nn/functional.py

```diff
@@ -565,3 +565,19 @@ def pad(input, pad, mode='constant', value=0):
             return _functions.thnn.ReplicationPad3d(*pad)(input)
     else:
         raise NotImplementedError("Only 4D and 5D padding is supported for now")
+
+
+# distance
+
+def pairwise_distance(input, p=2):
+    """Computes the p-norm between two vectors
+
+    Args
+        input (Variable): Tensor with size (2, batch, dim)
+        p (int) default=2: The norm degree
+    """
+    assert len(input) == 2, "Input needs to be of size (2, batch, dim)"
+    x1, x2 = input
+    diff = torch.abs(x1 - x2)
+    out = torch.pow(diff, p).sum(dim=1)
+    return torch.pow(out + 1e-6, 1. / p)
```

Comment:
Not sure how common it is to use `p < 1`, but for these values its the first `pow` that will be numerically unstable and the eps needs to be added there instead.